addon/client.go

```diff
@@ -179,6 +191,57 @@ func (r *Client) Delete(path string) (err error) {
 	return
 }
 
+//
+// Send the request.
+// Resilient against transient hub availability.
+// Retries for 10 minutes.
+func (r *Client) send(rb func() (*http.Request, error)) (response *http.Response, err error) {
+	var request *http.Request
+	err = r.buildTransport()
+	if err != nil {
+		return
+	}
+	for i := 0; i < 60; i++ {
+		request, err = rb()
+		if err != nil {
+			return
+		}
+		client := http.Client{Transport: r.transport}
+		response, err = client.Do(request)
+		netErr := &net.OpError{}
```

Comment:
Just a note on the Network Error type, If I understand it correctly, it covers cases with unreachable network, timeout etc., not HTTP errors like 4xx or 5xx - If this is right I assume it was intended.

Reply:
Exactly.